app/src/main/java/org/simple/clinic/teleconsultlog/drugduration/DrugDurationSheet.kt

```diff
@@ -0,0 +1,156 @@
+package org.simple.clinic.teleconsultlog.drugduration
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import android.view.inputmethod.EditorInfo
+import com.jakewharton.rxbinding3.widget.editorActions
+import io.github.inflationx.viewpump.ViewPumpContextWrapper
+import io.reactivex.Observable
+import io.reactivex.rxkotlin.ofType
+import kotlinx.android.synthetic.main.sheet_drug_duration.*
+import org.simple.clinic.ClinicApp
+import org.simple.clinic.R
+import org.simple.clinic.ReportAnalyticsEvents
+import org.simple.clinic.di.InjectorProviderContextWrapper
+import org.simple.clinic.drugs.PrescribedDrug
+import org.simple.clinic.mobius.MobiusDelegate
+import org.simple.clinic.teleconsultlog.drugduration.di.DrugDurationComponent
+import org.simple.clinic.util.LocaleOverrideContextWrapper
+import org.simple.clinic.util.unsafeLazy
+import org.simple.clinic.util.wrap
+import org.simple.clinic.widgets.BottomSheetActivity
+import org.simple.clinic.widgets.setTextAndCursor
+import org.simple.clinic.widgets.textChanges
+import java.util.Locale
+import javax.inject.Inject
+
+class DrugDurationSheet : BottomSheetActivity(), DrugDurationUiActions {
+
+  @Inject
+  lateinit var locale: Locale
+
+  @Inject
+  lateinit var effectHandlerFactory: DrugDurationEffectHandler.Factory
+
+  companion object {
+    private const val EXTRA_DRUG = "prescribedDrug"
+    private const val EXTRA_DURATION = "drugDuration"
+    private const val EXTRA_SAVED_DURATION = "savedDrugDuration"
+
+    fun intent(
+        context: Context,
+        drug: PrescribedDrug,
+        drugDuration: String
+    ): Intent {
+      return Intent(context, DrugDurationSheet::class.java).apply {
+        putExtra(EXTRA_DRUG, drug)
```

Comment:
Do we need the entire drug object in the extras? We can just use the `UUID`, correct?

Reply:
Since this mainly for setting the title and dosage I wasn't sure if we should do another DB query in this screen and save it to model. WDYT?

Reply:
Ah, we can pass the entire model in the factory function and then put only the values we need as part of the arguments.